go/vault/api/vault.go

```diff
@@ -0,0 +1,137 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/oasisprotocol/oasis-core/go/common/prettyprint"
+	staking "github.com/oasisprotocol/oasis-core/go/staking/api"
+)
+
+var _ prettyprint.PrettyPrinter = (*Authority)(nil)
+
+// State is the vault state.
+type State uint8
+
+const (
+	StateActive    = 0
+	StateSuspended = 1
+)
+
+// Vault contains metadata about a vault.
+type Vault struct {
+	// CreatedBy is the address of the vault creator.
+	CreatedBy staking.Address `json:"created_by"`
+	// CreatedNonce is the nonce of the vault creator's account at time of creation.
+	CreatedNonce uint64 `json:"created_nonce"`
```

Comment:
Yes the initial implementation had autoincrementing identifiers, but this is bad because all identifiers are predictable in advance. We could also go with block height and creation transaction index, but that lacks entropy. We could add a salt but that is awkward. I think the creator address and nonce works nicely.

Reply:
Oh, now I understand why you used metadata in comments. Well, I would still not expect this metadata in the `Vault` struct, just fields `Address`, `Nonce`, `State`, ...